### ISSUE TITLE
there's a circular dependency

### ISSUE THREAD

**Issue by mreinstein** [IjzerenHein/kiwi.js#16]
`expression.ts` -> `variable.ts` -> `expression.ts`

**Comment by mreinstein** [comment]
which is causing errors in bundling via rollup
![screen shot 2019-02-22 at 1 35 28 pm](https://user-images.githubusercontent.com/718067/53272673-c4753f00-36a6-11e9-8784-9d56be1617d2.png)

**Comment by IjzerenHein** [comment]
I know, it's been in there for a long time. Is it causing problems for you?

**Comment by mreinstein** [comment]
actually it appears that rollup is able to deal with this, it isn't causing breakage. It would be a nice design thing to fix, but not nearly as critical as I initially thought.

**Comment by IjzerenHein** [comment]
Agreed. I Checked the code but couldn't find a quick solution. I guess it would require merging some files to resolve this issue, but I doubt that is what we want.

**Comment by leeoniya** [comment]
see https://github.com/rollup/rollup/issues/2271 for workaround

**Comment by adamhaile** [comment]
My understanding is that circular dependencies are a warning in rollup (yellow), not an error (red), because in a lot of cases they're fine.  For instance, they're touted by [the rollup docs](https://rollupjs.org/guide/en#faqs) as a feature that es6 modules have over CommonJS.  Rollup can't (yet) distinguish the ok cases from the not ok, so it currently always warns.

The usage here, between `expression.ts` and `variable.ts`, looks fine to me.  I propose closing?

**Comment by mreinstein** [comment]
>  I Checked the code but couldn't find a quick solution.

@IjzerenHein we _could_ eliminate the circular dependency by moving the `Variable.plus`, `Variable.minus`, `Variable.multiply`, and `Variable.divide` functions into `Expression`.  Those are the only references to expressions from with the variable module. And since they are creating expressions it might be reasonable to put them there anyway. perhaps the function prototype changes from

```typescript
minus( value: number|Variable|Expression ): Expression
```

to
```typescript
minus( value: number|Variable|Expression,  value2: number|Variable|Expression ): Expression`)
```

I realize this might be controversial as it would be a major api semver bump for the sake of fixing a non-critical bundling error, but it wouldn't be hard to do this.
